core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -149,12 +151,15 @@ class LogManager(val logDirs: Array[File],
           val logRecoveryPoint = recoveryPoints.getOrElse(topicPartition, 0L)
 
           val current = new Log(logDir, config, logRecoveryPoint, scheduler, time)
-          val previous = this.logs.put(topicPartition, current)
-
-          if (previous != null) {
-            throw new IllegalArgumentException(
-              "Duplicate log directories found: %s, %s!".format(
-              current.dir.getAbsolutePath, previous.dir.getAbsolutePath))
+          if (logDir.getName.endsWith(Log.DeleteDirSuffix)) {
```

Comment:
If a particular broker holds a large amount of data for a particular topic or if we have a high throughput topic that was marked for delete, then deleting them synchronously when the broker starts up might increase the startup time significantly.